### ISSUE TITLE
A black box as a result

### ISSUE THREAD

**Issue by Weaita** [kijai/ComfyUI-LivePortraitKJ#34]
What is seen in the image, I am using comfyui portable

![imagen](https://github.com/kijai/ComfyUI-LivePortraitKJ/assets/17994944/d9681a7a-c8f5-411c-90e0-55bae884e600)
![imagen](https://github.com/kijai/ComfyUI-LivePortraitKJ/assets/17994944/80a8d1ba-aa34-4942-968a-f2676d5000d8)

**Comment by DavidSnow1** [comment]
This is an animation toolset, and you are using two static images. 

Try using an actual video, or an image sequence. Plug that into the 'driving images' socket.

**Comment by Weaita** [comment]
> This is an animation toolset, and you are using two static images.
> 
> Try using an actual video, or an image sequence. Plug that into the 'driving images' socket.

same here, using example workflow
![imagen](https://github.com/kijai/ComfyUI-LivePortraitKJ/assets/17994944/6b6b8093-19f9-4306-b4e0-f452a8f711e7)

**Comment by xox1234reddit** [comment]
I'm getting the exact same issue - black box over the face, using a mp4 video and a source image.

**Comment by brayevalerien** [comment]
Black images are usually produced by NaNs in tensors. Is your Pytorch up to date?

**Comment by FNSpd** [comment]
It's because of dtype. If you replace every "dtype=torch.float16" in live_portrait_wrapper.py with "dtype=torch.float32", it will work. Automatic switch of dtype should be implemented for systems that don't support it

**Comment by kijai** [comment]
What system is everyone getting black outputs using then?

**Comment by FNSpd** [comment]
> What system is everyone getting black outputs using then?

I was getting this error with GTX 1650, which can't perform FP16 calculations properly, so this was quite obvious what's the reason of the problem. Not sure about OP

**Comment by kijai** [comment]
> > What system is everyone getting black outputs using then?
> 
> I was getting this error with GTX 1650, which can't perform FP16 calculations properly, so this was quite obvious what's the reason of the problem. Not sure about OP

Ok, I've added the choice to the model loader now.

**Comment by xox1234reddit** [comment]
Tried the updated nodes, still getting a black box on portable

**Comment by xox1234reddit** [comment]
> > Tried the updated nodes, still getting a black box on portable
> 
> With fp32 selected?

Where would I select that? Forgive me, I thought this would be automatic, so I didn't look for a selection on my end

**Comment by kijai** [comment]
> > > Tried the updated nodes, still getting a black box on portable
> > 
> > 
> > With fp32 selected?
> 
> Where would I select that? Forgive me, I thought this would be automatic, so I didn't look for a selection on my end

It's one of the problems when updating nodes, you have to refresh the page if the node existed in an old workflow. It should appear in the loader node:

![image](https://github.com/kijai/ComfyUI-LivePortraitKJ/assets/40791699/7f91cccf-d40e-4773-a87f-afe6ee38d462)

**Comment by xox1234reddit** [comment]
> > > > Tried the updated nodes, still getting a black box on portable
> > > 
> > > 
> > > With fp32 selected?
> > 
> > 
> > Where would I select that? Forgive me, I thought this would be automatic, so I didn't look for a selection on my end
> 
> It's one of the problems when updating nodes, you have to refresh the page if the node existed in an old workflow. It should appear in the loader node:
> 
> ![image](https://private-user-images.githubusercontent.com/40791699/346655541-7f91cccf-d40e-4773-a87f-afe6ee38d462.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA0NTg5MjcsIm5iZiI6MTcyMDQ1ODYyNywicGF0aCI6Ii80MDc5MTY5OS8zNDY2NTU1NDEtN2Y5MWNjY2YtZDQwZS00NzczLWE4N2YtYWZlNmVlMzhkNDYyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA4VDE3MTAyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNlMDE0M2FhZjQ4NWRiZGRmZGE5NGE0NWVmNWY3ODgzZDNjMzNjMjliOTgzMDQxMWJhMGU5YWJhNWUzODhkNTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NhkbMFbi3PT8JmJguZvQPcX2mmZmfVN4Y5uAugu2TfQ)

FINALLY got it to appear, but now I have to load the default workspace, hit refresh, and then load up my saved workspace or else the mouse will not work and my workspace looks blank. Let me test the new feature to see if this fixes issues

**Comment by xox1234reddit** [comment]
Okay, so now the video result is NOWHERE NEAR the driving video. It looks like it just dumped every single possible expression into one video.

https://github.com/kijai/ComfyUI-LivePortraitKJ/assets/174985240/fe3d726b-7af3-4e06-88c3-8fb5f2170bae

**Comment by Weaita** [comment]
I tested the latest version and now it works. By the way, I have a GTX 1650
Thank you very much for the solution!! :)
![imagen](https://github.com/kijai/ComfyUI-LivePortraitKJ/assets/17994944/3a901d67-14be-4125-8e9d-0931386c36e8)

**Comment by kijai** [comment]
> Okay, so now the video result is NOWHERE NEAR the driving video. It looks like it just dumped every single possible expression into one video.
> 
> Doing a second test with a different driving video to see if it improves
> 
> EDIT - It's DEFINITELY a driving video issue lol swapped it out and seems to work fine
> 
>  AnimateDiff_00001.mp4

Yes the driving video currently has to be cropped, the included example driving vids work as reference of what works well.

**Comment by kijai** [comment]
> @kijai  Is there an option to choose or show single subject in the final output instead of showing the comparison side by side with input video?

Just connect one of the outputs from the Process node directly to the video combine.

**Comment by amit4aws** [comment]
> > @kijai  Is there an option to choose or show single subject in the final output instead of showing the comparison side by side with input video?
> 
> Just connect one of the outputs from the Process node directly to the video combine.

Thanks also is it only supposed to be run on local machine? which this nodes in place If i start comfyUI it is always getting me the URL to access in local host and not remote. My workspace is on cloud vm and so can not use it with localhost url

[http://127.0.0.1:6006](javascript:void(0);)
[https://127.0.0.1:6007](javascript:void(0);)

I even have the --listen arg set for comfy ui when i am running it on cloud vm. If i uninstall this nodes then everything is back to normal and i can access my comfui with remote url.

**Comment by kijai** [comment]
> > > @kijai  Is there an option to choose or show single subject in the final output instead of showing the comparison side by side with input video?
> > 
> > 
> > Just connect one of the outputs from the Process node directly to the video combine.
> 
> Thanks also is it only supposed to be run on local machine? which this nodes in place If i start comfyUI it is always getting me the URL to access in local host and not remote. My workspace is on cloud vm and so can not use it with localhost url
> 
> http://127.0.0.1:6006 https://127.0.0.1:6007
> 
> I even have the --listen arg set for comfy ui when i am running it on cloud vm. If i uninstall this nodes then everything is back to normal and i can access my comfui with remote url.

I don't understand how a specific custom node could even affect that, if you use --listen the ip should be 0.0.0.0. I have tons of custom nodes installed and it works fine with --listen.
